File: FreakFightsFan.Shared/Features/Fighters/Queries/GetAllFighters.cs
using FluentValidation;
using FreakFightsFan.Shared.Abstractions;
using FreakFightsFan.Shared.Features.Fighters.Responses;
using MediatR;

namespace FreakFightsFan.Shared.Features.Fighters.Queries;

public static class GetAllFighters
{
    public class Query : IRequest<PagedList<FighterDto>>, IPagedQuery, ISortedQuery
    {
        public string SearchTerm { get; set; }

        public List<int> HiddenFightersIds { get; set; }
        public int Page { get; set; }
        public int PageSize { get; set; }
        public string SortColumn { get; set; }
        public SortOrder SortOrder { get; set; }
    }

    public class Validator : AbstractValidator<Query> { }
}

File: freak-fights-fan/shared/features/fighters/queries/GetAllFighters.ts
import { SortOrder } from "../../../abstractions/SortOrder";
import { IPagedQuery } from "../../../abstractions/IPagedQuery";
import { ISortedQuery } from "../../../abstractions/ISortedQuery";

export namespace GetAllFighters {
    export interface Query extends IPagedQuery, ISortedQuery {
        page: number;
        pageSize: number;
        sortColumn: string;
        sortOrder: SortOrder;
        searchTerm: string;

        hiddenFightersIds: Array<number>;
    }
}
